wallet-new/src/Cardano/Wallet/Client.hs

```diff
@@ -0,0 +1,126 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE StrictData #-}
```

Comment:
First time I see this pragma used in the wild -- nice! (I remember when I was a naive Haskeller and wanted it badly to strictify-all-the-things 😁 )

Reply:
Strict record fields = no space leaks, in my experience :)

I mostly want them because then you et a compile time error if you don't initialize all the fields in a record. Doing:

```haskell
data X = X { x :: Int, y :: Int }

oops = y (X { x = 3 })
```

gets a runtime error, while

```haskell
data X = X { x :: !Int, y :: !Int }

oops = X { x = 3 }
```

gets you a compile time error.

Reply:
oh, TIL regarding the runtime/compile error, thanks!

> Strict record fields = no space leaks, in my experience :)

Yea, pretty much my experience as well. There might be good reasons (sometimes) to not "bang" fields in records, but it's 90% what you want, in practice, at least if your record has very simply "primitive" types.